components/aws/sagemaker/common/_utils.py

```diff
@@ -471,11 +471,61 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  endpoint_exists = False
+  if endpoint_name:
+      endpoints = client.list_endpoints(
+          MaxResults=99,
+          NameContains=endpoint_name,
+      )['Endpoints']
+      for i in range(len(endpoints)):
+          ep = endpoints[i]
+          if (ep['EndpointName'] == endpoint_name):
+            endpoint_exists = True
+            logging.info("Endpoint exists: " + ep['EndpointName'] + " - " + ep['EndpointArn'])
+  return endpoint_exists
+
+def endpoint_config_name_exists(client, endpoint_config_name):
+  ep_config_name_exists = False
+  if endpoint_config_name:
+      if client.describe_endpoint_config(
+          EndpointConfigName=endpoint_config_name,
+          )['EndpointConfigName'] == endpoint_config_name:
+        ep_config_name_exists = True
+        logging.info("Endpoint Config exists: " + endpoint_config_name)
+  return ep_config_name_exists
+
 def deploy_model(client, args):
-  endpoint_config_name = create_endpoint_config(client, args)
-  endpoint_name = create_endpoint(client, args['region'], args['endpoint_name'], endpoint_config_name, args['endpoint_tags'])
+  if args['update_endpoint'] and endpoint_name_exists(client, args['endpoint_name']):
+      endpoint_config_name = create_endpoint_config(client, args)
+      endpoint_name = update_endpoint(client, args['region'], args['endpoint_name'], endpoint_config_name)
+  else:
+      args['update_endpoint'] = False
+      endpoint_config_name = create_endpoint_config(client, args)
+      endpoint_name = create_endpoint(client, args['region'], args['endpoint_name'], endpoint_config_name, args['endpoint_tags'])
   return endpoint_name
 
+def get_current_endpoint_config(client, args):
+    endpoint_config_name = ''
```

Comment:
nit: Empty string value should be `None` rather than `''`. This makes it easier to test for, we can do `if not <string>:`.

Reply:
Done